src/Components/3DV/SceneView.tsx

```diff
@@ -1673,6 +1680,212 @@ function SceneView(props: ISceneViewProps, ref) {
+                            gizmoTransformItemDraftRef.current.transform
+                        );
+                    });
+
+                    // on drag end for both position and rotation gizmos, update transform
+                    // updating on every frame is too much for react to handle
+                    const positionGizmo = gizmoManager.gizmos.positionGizmo;
+                    positionGizmo.onDragEndObservable.add(() => {
+                        // update the gizmoTransformItem (allows builder panel to display new position/rotation)
+                        const attachedMesh =
+                            gizmoManager.gizmos.positionGizmo.attachedMesh;
+
+                        // this is the main place where transforms get set, so round here
+                        gizmoTransformItemDraftRef.current.transform.position = {
+                            x: Math.round(attachedMesh.position.x),
+                            y: Math.round(attachedMesh.position.y),
+                            z: Math.round(attachedMesh.position.z)
+                        };
+                        setGizmoTransformItem(
+                            gizmoTransformItemDraftRef.current.transform
+                        );
+                    });
+
+                    const rotationGizmo = gizmoManager.gizmos.rotationGizmo;
+                    rotationGizmo.onDragEndObservable.add(() => {
+                        // update the gizmoTransformItem (allows builder panel to display new position/rotation)
+                        const attachedMesh =
+                            gizmoManager.gizmos.positionGizmo.attachedMesh;
+
+                        // this is the main place where transforms get set, so round here
+                        gizmoTransformItemDraftRef.current.transform.rotation = {
+                            x: Number(attachedMesh.rotation.x.toFixed(2)),
+                            y: Number(attachedMesh.rotation.y.toFixed(2)),
+                            z: Number(attachedMesh.rotation.z.toFixed(2))
+                        };
+                        setGizmoTransformItem(
+                            gizmoTransformItemDraftRef.current.transform
+                        );
+                    });
+                }
+            } catch {
+                console.warn('unable to add gizmo to mesh');
+            }
+            if (debugLogging) {
+                console.timeEnd('adding gizmo to meshes');
+            }
+        }
+
+        return () => {
+            gizmoManagerRef.current?.dispose();
+            gizmoManagerRef.current = null;
+            // only cleanup if exiting edit behavior, not when switching tabs
+            if (!gizmoElementItem) {
+                gizmoTransformItemDraftRef.current = null;
+            }
+            debugLog('debug', 'Mesh gizmo cleanup');
+        };
+    }, [scene, gizmoElementItem, isLoading]);
+
+    // Handle gizmoTransformItem
+    useEffect(() => {
```

Comment:
so if I understand correctly, this is for specifically handling the text box inputs in the left panel. Is there a reason that this requires a separate set of state? Could we pass in the position of the object the same way for the manual inputs as well as the click and drag?